.github/workflows/onRelease.yml

```diff
@@ -2,18 +2,31 @@ name: publish
 
 on:
   release:
-    types: [released]
+    types: [published]
   # support manual release in case something goes wrong and needs to be repeated or tested
   workflow_dispatch:
     inputs:
       tag:
         description: tag that needs to publish
         type: string
         required: true
+
 jobs:
+  getDistTag:
+    outputs:
+      tag: ${{ steps.distTag.outputs.tag }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.release.tag_name || inputs.tag  }}
+      - uses: oclif/github-workflows/.github/actions/getPreReleaseTag@prereleases-semver
+        id: distTag
+
   npm:
     uses: oclif/github-workflows/.github/workflows/npmPublish.yml@main
+    needs: [getDistTag]
     with:
-      tag: latest
+      tag: ${{ needs.getDistTag.outputs.tag || 'latest' }}
```

Comment:
This shouldn't default to using `latest` because the `getPreReleaseTag` action can return an empty or falsy value when a prerelease is intended.